### PR TITLE
Use redirect unpublishing

### PR DESCRIPTION
**Pull request by cbaines**
Just putting this up before I forget.

### CODE REVIEW

**Comment by chao-xian** [comment]
@cbaines did you forget? 😄

**Comment by cbaines** [comment]
> @cbaines did you forget? 😄

The forgetting was more about about publishing anything.

I think this would be an improvement, but I haven't thought through it fully, or done any testing. Are you looking to retire more smart answers soon?

**Comment by chao-xian** [comment]
There is one that came out of Firebreak to convert it to a Simple Smart Answer (which I think is a format in publisher). We're waiting on the Content Team to get green light from SME and once they do we'll retire the smart answer and re-point the base path to the new simple smart answer.

**Comment by chao-xian** [comment]
Would this actually help in the case I've just described?

**Comment by cbaines** [comment]
> Would this actually help in the case I've just described?

This change is mostly a neatness thing, it doesn't really help or hinder as far as I can remember.

File: lib/tasks/retire.rake [hunk 1]
```diff
@@ -7,8 +7,9 @@ namespace :retire do
 
     content_item_publisher = ContentItemPublisher.new
 
-    content_item_publisher.unpublish(args.content_id)
-    content_item_publisher.publish_redirect(args.base_path, args.destination)
+    content_item_publisher.unpublish_with_redirects(
+      args.content_id, args.path, args.destination
```
Comment (AlanGabbianelli) [inline lib/tasks/retire.rake]: Should `args.path` be `args.base_path`?

**Comment by bevanloon** [comment]
@cbaines is this still relevant or can we close the pr?

**Comment by cbaines** [comment]
> @cbaines is this still relevant or can we close the pr?

I believe it's still relevant, well, just for smart answer retiring, which doesn't happen very regularly.

File: lib/tasks/retire.rake (hunk 1 above)
Comment (cbaines) [reply lib/tasks/retire.rake]: Yep, good spot, I've updated the commit.

**Pull request merged**